// npkill/lib/services/files.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileService = void 0;
const fs = require("fs");
class FileService {
    convertKbToGB(kb) {
        const factorKBtoGB = 1048576;
        return kb / factorKBtoGB;
    }
    convertBytesToKB(bytes) {
        const factorBytestoKB = 1024;
        return bytes / factorBytestoKB;
    }
    convertGBToMB(gb) {
        const factorGBtoMB = 1024;
        return gb * factorGBtoMB;
    }
    getFileContent(path) {
        const encoding = 'utf8';
        return fs.readFileSync(path, encoding);
    }
    isSafeToDelete(path, targetFolder) {
        return path.includes(targetFolder);
    }
    isDangerous(path) {
        /* We consider a directory to be dangerous if it is hidden.
         
          > Why dangerous?
          It is probable that if the node_module is included in some hidden directory, it is
          required by some application like "spotify", "vscode" or "Discord" and deleting it
          would imply breaking the application (until the dependencies are reinstalled).
        */
        const hiddenFilePattern = /(^|\/)\.[^\/\.]/g;
        return hiddenFilePattern.test(path);
    }
}
exports.FileService = FileService;
