// npkill/lib/services/update.service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateService = void 0;
const update_constants_1 = require("@core/constants/update.constants");
class UpdateService {
    constructor(httpsService) {
        this.httpsService = httpsService;
    }
    async isUpdated(localVersion) {
        const remoteVersion = await this.getRemoteVersion();
        const local = this.splitVersion(localVersion);
        const remote = this.splitVersion(remoteVersion);
        return this.compareVersions(local, remote);
    }
    compareVersions(local, remote) {
        return (this.isSameVersion(local, remote) ||
            this.isLocalVersionGreater(local, remote));
    }
    async getRemoteVersion() {
        const response = await this.httpsService.get(update_constants_1.VERSION_CHECK_DIRECTION);
        return response[update_constants_1.VERSION_KEY];
    }
    splitVersion(version) {
        const versionSeparator = '.';
        const remoteSplited = version.split(versionSeparator);
        return remoteSplited.join('');
    }
    isSameVersion(version1, version2) {
        return version1 === version2;
    }
    isLocalVersionGreater(local, remote) {
        return local > remote;
    }
}
exports.UpdateService = UpdateService;
