// npkill/lib/strategies/windows-strategy.abstract.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WindowsStrategy = void 0;
const process_1 = require("process");
class WindowsStrategy {
    constructor() {
        const { major, minor } = this.getNodeVersion();
        this.major = major;
        this.minor = minor;
    }
    setNextStrategy(next) {
        this.next = next;
        return next;
    }
    checkNext(path, callback) {
        if (!this.next) {
            return true;
        }
        return this.next.remove(path, callback);
    }
    getNodeVersion() {
        const releaseVersionsRegExp = /^v(\d{1,2})\.(\d{1,2})\.(\d{1,2})/;
        const versionMatch = process_1.version.match(releaseVersionsRegExp);
        if (!versionMatch) {
            throw new Error(`Unable to parse Node version: ${process_1.version}`);
        }
        return {
            major: parseInt(versionMatch[1], 10),
            minor: parseInt(versionMatch[2], 10),
            patch: parseInt(versionMatch[3], 10),
        };
    }
}
exports.WindowsStrategy = WindowsStrategy;
